architecture.md

```diff
@@ -0,0 +1,143 @@
+# Architecture
+
+This document describes the high-level architecture of the State Tool and
+related applications (currently: State Service, State Tray, State Installer, and
+State Update Dialog).
+
+## Applications Overviews
+
+### State Tool
+
+The State Tool is a CLI application providing access to the ActiveState Platform
+features. It is the primary consumer of ActiveState Platform APIs which provide
+runtime, library and application builds. Using these reproducible, indemnified,
+and self-serve artifacts, the State Tool modifies the host environment in order
+to provide package and virtual environment management.
+
+### State Service
+
+The State Service provides the State Tool with a central point of access to the
+ActiveState Platform APIs, host environment modification, and other features
+like caching.
+
+### State Tray
+
+The State Tray provides a GUI element which helps to bring important information
+to the visual forefront of a user's experience (e.g. available updates, host
+environment status, etc.), as well as quick access launchers (e.g. ActiveState
+Platform Dashboard and support links, project entry points, etc.).
+
+### State Installer
+
+The State Installer provides a cross-platform installer for the suite of State
+Tool applications. Shell script wrappers are provided as a convenience.
+
+### State Update Dialog
+
+The State Update Dialog provides a cross-platform updater for the suite of State
+Tool Applications.
+
+## Directory Structure
+
+### assets/
+
+Various files (e.g. images, templates, etc.) used by any application.
+
+### build/
+
+Artifacts resulting from building applications.
+
+### cmd/*/
+
+Individual "main" applications.
+
+#### cmd/*/internal/
+
+Packages used exclusively for the parent application.
+
+### docs/
+
+Developer-focused documentation.
+
+### .github/workflows-src/
+
+YAML-formatted config that is processed by `ytt`
+(https://github.com/vmware-tanzu/carvel-ytt) to produce yml files used for CI
+(stored in `.github/workflows/`).
+
+### installers/
+
+Shell scripts wrapping the installer application for user convenience.
+
+### internal/
+
+Packages that are made available for use by any application, but are restricted
+from use by external code.
+
+#### internal/runbits/
+
+Packages that are made available for use by "runner" packages. In essence,
+`internal/runners/internal/runbits`.
+
+#### internal/runners/
+
+Packages that provide command behavior for the State Tool application.
+
+### locale/
+
+Localization keys and associated values.
+
+### pkg/
+
+Packages that are made available for use by any application, including external
+code. The packages that are currently located here are not intended for use by
+external code, and will eventually be moved into the `internal/` directory.
+
+#### pkg/cmdlets/
+
+Packages that are made available for use by "runner" packages. Synonymous with
+`internal/runbits/`, and all new runner-common packages should be placed there.
```

Comment:
Other way around; runbits is new, cmdlets are old.

Reply:
That is what I tried to convey, that runbits is the location of all new common packages. Should I word it differently? Maybe:
```suggestion
#### pkg/cmdlets/

Packages that are made available for use by "runner" packages. Synonymous with
`internal/runbits/` with all new runner-common packages being placed there.
```

Reply:
I think you should make the title `pkg/runbits` and instead hint at cmdlets. runbits are the way forward, so it should take center stage.

Reply:
That directory is covered: https://github.com/ActiveState/cli/pull/1556/files#diff-4e03c430fc7f575c72fd989625acdc7d8d70fd20d3693c893d67d9dbd461ccdbR77-R81

Reply:
Just pushed a change to remove the pkg/cmdlets entry and update the runbits entry.